hudi-sync/hudi-sync-common/src/main/java/org/apache/hudi/sync/common/metrics/HoodieMetaSyncMetrics.java

```diff
@@ -0,0 +1,112 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.sync.common.metrics;
+
+import org.apache.hudi.common.util.StringUtils;
+import org.apache.hudi.common.util.VisibleForTesting;
+import org.apache.hudi.config.metrics.HoodieMetricsConfig;
+import org.apache.hudi.hadoop.fs.HadoopFSUtils;
+import org.apache.hudi.metrics.Metrics;
+import org.apache.hudi.storage.HoodieStorage;
+import org.apache.hudi.storage.HoodieStorageUtils;
+import org.apache.hudi.sync.common.HoodieSyncConfig;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Timer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HoodieMetaSyncMetrics {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HoodieMetaSyncMetrics.class);
+  // Metrics are shut down by the shutdown hook added in the Metrics class
+  private Metrics metrics;
+  private HoodieMetricsConfig metricsConfig;
+  private transient HoodieStorage storage;
+
+  private final String syncToolName;
+
+  private String recreateAndSyncFailureCounterName;
+  private String recreateAndSyncTimerName;
+
+  private Timer recreateAndSyncTimer;
+  private Counter recreateAndSyncFailureCounter;
+
+  public HoodieMetaSyncMetrics(HoodieSyncConfig config, String syncToolName) {
+    this.metricsConfig = config.getMetricsConfig();
+    this.syncToolName = syncToolName;
+    if (metricsConfig.isMetricsOn()) {
+      this.storage = HoodieStorageUtils.getStorage(config.getBasePath(), HadoopFSUtils.getStorageConf(config.getHadoopConf()));
+      metrics = Metrics.getInstance(metricsConfig, storage);
+      this.recreateAndSyncTimerName = getMetricsName("timer", "meta_sync.recreate_table");
+      this.recreateAndSyncFailureCounterName = getMetricsName("counter", "meta_sync.recreate_table.failure");
+    }
+  }
+
+  public Metrics getMetrics() {
+    return metrics;
+  }
+
+  public Timer.Context getRecreateAndSyncTimer() {
+    if (metricsConfig.isMetricsOn() && recreateAndSyncTimer == null) {
+      recreateAndSyncTimer = createTimer(recreateAndSyncTimerName);
+    }
+    return recreateAndSyncTimer == null ? null : recreateAndSyncTimer.time();
+  }
+
+  private Timer createTimer(String name) {
+    return metricsConfig.isMetricsOn() ? metrics.getRegistry().timer(name) : null;
+  }
+
+  public void incrementRecreateAndSyncFailureCounter() {
+    recreateAndSyncFailureCounter = getCounter(recreateAndSyncFailureCounter, recreateAndSyncFailureCounterName);
+    recreateAndSyncFailureCounter.inc();
+  }
+
+  public void updateRecreateAndSyncDurationInMs(long durationInNs) {
+    if (metricsConfig.isMetricsOn()) {
+      long durationInMs = getDurationInMs(durationInNs);
+      LOG.info("Sending recreate and sync metrics {}", durationInMs);
+      metrics.registerGauge(getMetricsName("meta_sync", "recreate_table.duration"), durationInMs);
```

Comment:
Same for this metric name.  Also, rename it with time unit `recreate_table_duration_ms`